packages/frontend-asset-loader/src/utils/utils.ts

```diff
@@ -0,0 +1,55 @@
+import { PathAndFile } from "./utils.types";
+import { logger, CustomLogger } from "../utils/logger";
+
+export const getDuplicateHashedFileName = (
+  array: PathAndFile[],
+): string | null => {
+  const seen = new Set<string>();
+
+  for (const obj of array) {
+    if (seen.has(obj.hashedFileName)) {
+      return obj.hashedFileName;
+    }
+    seen.add(obj.hashedFileName);
+  }
+
+  return null;
+};
+
+export const isValidHashName = (
+  hashName: string,
+  hashStart: string,
+  hashEnd: string,
+  customLogger?: CustomLogger,
```

Comment:
I don't think this argument is provided when this function is called. To avoid having to drill the logger down in to these functions it's probably worth turning the logger in to a Singleton that we set at the entry-point to the package, and get any time we need to access it. Sorry—happy to talk through what I'm getting at here if it's not clear.